presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java

```diff
@@ -977,4 +994,10 @@ public HiveConfig setProjectionPushdownEnabled(boolean projectionPushdownEnabled
         this.projectionPushdownEnabled = projectionPushdownEnabled;
         return this;
     }
+
+    @PostConstruct
+    public void validate()
+    {
+        InsertExistingPartitionsBehavior.validate(insertExistingPartitionsBehavior, immutablePartitions);
```

Comment:
Will we deprecate immutable partitions property?

Reply:
No. `immutable partitions = true` is used to protect setting `hive.insert-existing-partitions-behavior` to APPEND state via session properties.

Reply:
You could extend `hive.insert-existing-partitions-behavior` with `FORCE_ERROR` (name subject to discussion) which behaves as ERROR, but also be non overridable by session property.

Reply:
For some reason `immutable partitions=true` allows setting `insert-existing-partitions-behavior=OWERWRITE` that's why it is not an option. Apart from it breaks backwards compatibility and requires longer discussion. 
Won't do.

Reply:
If I read the code correctly now if we specify `hive.immutable-partitions=true` and NOT specify `hive.insert-existing-partitions-behavior` the Presto will fail to start.
This is valid setup today. To be consistent with what we have right now we should default `hive.insert-existing-partitions-behavior` to `ERROR` if `hive.immutable-partitions` is `true`.

Reply:
Also can you add test for above.

Reply:
Good point!

I've tried to calculate default value during @PostConstruct phase
```
    private Optional<InsertExistingPartitionsBehavior> insertExistingPartitionsBehavior = Optional.empty();

    public InsertExistingPartitionsBehavior getInsertExistingPartitionsBehavior()
    {
        return insertExistingPartitionsBehavior.orElseThrow(() -> new NoSuchElementException("insertExistingPartitionsBehavior was not set during presto initialization"));
    }

    @Config("hive.insert-existing-partitions-behavior")
    @ConfigDescription("Default value for insert existing partitions behavior")
    public HiveConfig setInsertExistingPartitionsBehavior(InsertExistingPartitionsBehavior insertExistingPartitionsBehavior)
    {
        this.insertExistingPartitionsBehavior = Optional.of(requireNonNull(insertExistingPartitionsBehavior, "insertExistingPartitionsBehavior is null"));
        return this;
    }

    @PostConstruct
    public void validate()
    {
        if (insertExistingPartitionsBehavior.isEmpty()) {
            insertExistingPartitionsBehavior = getInsertExistingPartitionsBehaviorForBackwardCompatibility();
        }
        else {
            InsertExistingPartitionsBehavior.validate(insertExistingPartitionsBehavior.get(), immutablePartitions);
        }
    }

    private Optional<InsertExistingPartitionsBehavior> getInsertExistingPartitionsBehaviorForBackwardCompatibility()
    {
        return Optional.of(immutablePartitions ? ERROR : APPEND);
    }
```
but seems it's not good idea from frameworks' point of view (even not looking to even more complicated logic), because tests are not configured to maintain such complex logic TestHiveConfig#testDefaults() does not invoke @PostConstruct method.

@kokosing there are two possible solutions:
1. apply solution proposed by @losipiuk which changes default value hive.insert-existing-partitions-behavior to ERROR, which breaks backward compatibility when immutablePartitions=false (in this case insert-existing-partitions-behavior was set to APPEND)
2. Break backwards compatibility for one of valid setups today: hive.immutable-partitions=true and hive.insert-existing-partitions-behavior is NOT specified.
What solution would be better?

Reply:
I'd implement solution 3.